opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+        ypy = np.average( y * uy, weights=w )
+        # Calculate the beam emittances
+        emit_x = np.sqrt( xsq * uxsq - xpx ** 2 )
+        emit_y = np.sqrt( ysq * uysq - ypy ** 2 )
+        # Return the results
+        return( emit_x, emit_y )
+
+    def get_current( self, t=None, iteration=None, species=None, select=None,
+                     bins=100 ):
+        """
+        Calculate the electric current along the z-axis for selected particles.
+
+        Parameters
+        ----------
+         t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        bins : float
+            Number of bins along the z-axis in which to calculate the current
+
+        Returns
+        -------
+        A tuple of arrays containig
+        - The current in each bin in Ampere
+        - The z positions of the bin edges
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        z, uz, uy, ux, w, q = self.get_particle(
+                               var_list=['z', 'uz', 'uy', 'ux', 'w', 'charge'],
+                               t=t, iteration=iteration,
+                               species=species )
+        # Calculate Lorentz factor for all particles
+        gamma = np.sqrt(1 + ux ** 2 + uy ** 2 + uz ** 2)
+        # Length to be seperated in bins
+        len_z = np.max(z) - np.min(z)
+        # Define bins
+        bins_list = np.linspace(np.min(z), np.max(z), bins)
+        # Get bin index of each particle
+        particle_bin = np.digitize(z, bins_list)
+        # Calculate sum of vz in each bin
+        vz_sum = np.zeros_like(bins_list)
+        for i in range(bins):
+            vz_sum[i] = np.sum(uz[particle_bin == i] / gamma[particle_bin == i]
+                               * const.c * w[particle_bin == i] )
+        # Calculete the current in each bin
+        current = vz_sum * np.abs(q) * bins / (len_z * 1.e-6)
+
+        # Return the current and bin edges
+        return(current, bins_list)
+
+    def get_laser_envelope( self, t=None, iteration=None, pol=None, m='all',
+                            freq_filter=40, index='center' ):
```

Comment:
However, I'm not quite sure about giving `theta` and `slicing_dir` as optionals, since they are implicitly determined by `pol`. So we should probably agree on one option.